api/controllers/ExportController.js

```diff
@@ -0,0 +1,136 @@
+    }
+    if (!p.datasets || !p.datasets.length) {
+      throw new Error("Please specify datasets to export")
+    }
+
+    // auth check
+    let ok = false
+    try {
+      ok = GroupMembership.hasModeratorRole(req.session.userId, p.groupId)
+    } catch (err) {
+      return res.status(422).send({ error: err.message ? err.message : err })
+    }
+
+    if (!ok) {
+      return res.status(403).send({ error: "No access" })
+    }
+
+    // process specified datasets
+    if (p.datasets.includes('members')) {
+      return exportMembers(p.groupId, req, res)
+    }
+
+    // got to the end and nothing output/exited, throw error
+    throw new Error("Unknown datasets specified: " + JSON.stringify(p.datasets))
+  }
+}
+
+/**
+ * Group members export by Group ID
+ */
+async function exportMembers(groupId, req, res) {
+  const users = await new Group({ id: groupId })
+                  .members()
+                  .fetch()
+
+  const results = []
+
+  // iterate over all group members
+  await Promise.all(users.map((u, idx) => {
+    // pluck core user data into results
+    results.push(u.pick([
+      'name', 'contact_email', 'contact_phone', 'location', 'avatar_url', 'tagline', 'bio',
+      'url', 'twitter_url', 'facebook_url', 'linkedin_url'
+    ]))
+
+    // return combined promise to load all dependent user data and
+    // assign final child query results back onto matching result objects upon completion
+    return Promise.all([
+
+      // affiliations
+      u.affiliations().fetch()
+        .then(affils => {
+          results[idx].affiliations = accumulatePivotCell(affils, renderAffiliation)
+        }),
+
+      // skills
+      u.skills().fetch()
+        .then(skills => {
+          results[idx].skills = accumulatePivotCell(skills, renderSkill)
+        }),
+
+      // skills to learn
+      u.skillsToLearn().fetch()
+        .then(skills => {
+          results[idx].skills_to_learn = accumulatePivotCell(skills, renderSkill)
+        }),
+
+      // Join request questions & answers
+      u.joinRequests().where({ group_id: groupId, status: JoinRequest.STATUS.Accepted }).fetch()
+        .then(jrs => Promise.all(jrs.map(jr =>
+          jr.questionAnswers().fetch()
+            .then(qas => Promise.all(qas.map(qa =>
+              Promise.all([
+                qa.question().fetch(),
+                Promise.resolve(qa)
+              ])
+            )))
+        )))
+        .then(data => {
+          results[idx].join_request_questions = accumulatePivotCell(data && data[0] || [], renderJoinRequestAnswer)
+        })
+
+    ])
+  }))
+
+  // send data as CSV response
+  output(res, results, [
+    'name', 'contact_email', 'contact_phone', 'location', 'avatar_url', 'tagline', 'bio',
+    { key: 'url', header: 'personal_url' },
+    'twitter_url', 'facebook_url', 'linkedin_url',
+    'skills', 'skills_to_learn',
+    'join_request_questions',
+    // :TODO: groups the person is a part of (that the moderator can see)
```

Comment:
the easiest way to do this is to load the user.memberships() and then apply the groupFilter from filters.js which filters out groups that the current user cant see. i think the way to do this is call `groupFilter(userId)(user.memberships()).fetch()` ? that just might do it